Independent views upon click on the 3 different panels of the app

Bachibouzouk opened the pull request:
This PR replaces #94 

Closes #91 and closes #93 

The NUTS records in the OEDB are for small landkreis. @christian-rli found a way to find the small NUTS from the NUTS of a larger region in #93. We could either do the OEP quey with a "look for text" in every row of the table, or use the nuts geojson @christian-rli prepared and use the `.in_()` sqlalchemy function to find records within that list.

Bachibouzouk committed:
Prevent unavailability of map.spin

Bachibouzouk committed:
Create dedicated layer for the regions

Bachibouzouk committed:
Create a dedicated layer for Landkreis

Bachibouzouk committed:
Create view function for landkreis

Bachibouzouk committed:
Update ajax query for feedinlib

The user can click on a landkreis and it will generate
a marker and fire the click on the marker automatically.
The click on the marker will trigger an OEP query to fetch
the information for this landkreis (currently the data will
only be from a dummy dataset)

Bachibouzouk committed:
Create dedicated layer for powerplants

Bachibouzouk committed:
Use nuts info to build OEP query

The ST_Transform function is not available in OEdialect
so we will fetch the powerplants in a region via their nuts
column in the OEDB instead.

Bachibouzouk committed:
Manage the different layers upon click on panel

Bachibouzouk committed:
Atom beautify and pylint

Bachibouzouk committed:
Merge develop into feature/independent-panels

Bachibouzouk commented:
@jh-RLI - the functionalities work with the LOCAL_TEST is True (i.e. no Query to OEDB via OEP).
I would recommend you try to figure out the query with the nuts and you fix the `serial_views.py`

jh-RLI committed:
Add class method nuts_1_for_region() it takes the ppr_region from f.e. post request and returns the nuts_1 code from geojson, added a serializer instance to ppr_view, get the nuts_1 code

jh-RLI committed:
fix pylint error

jh-RLI committed:
fix pylint error

jh-RLI commented:
@Bachibouzouk nuts work now for power plant register. Again im not sure why i get a pylint error with unused variable i.

jh-RLI reviewed (approved):
good

Bachibouzouk committed:
Rename geojson for region and landkreis

Bachibouzouk committed:
Replace function by a dict

each nuts_1 form a region correspond to a list of nuts_3
of the landkreis contained in theis region

Bachibouzouk committed:
Add an option to the fred config

This allows to choose to connect to the database via OEP API
(OEP_DIALECT) or via OEDB users (OEP) as some functions from
SQLAlchemy are not available via OEP API

Bachibouzouk committed:
Implement search of powerplants with nuts

The srid of the rea_geom_new is not specified in the case of
OEP_DIALECT, so geom is used instead to display points

Bachibouzouk committed:
Fix wrong function call

Bachibouzouk commented on app_settings.py:
```diff
@@ -1,23 +1,27 @@
 # add local config file
 fred_config = ConfigObj(os.path.join(settings.BASE_DIR, 'WAM_APP_FRED', 'config', 'fred_app.cfg'))
 
+OEP_ACCESS = fred_config['WAM_APP_FRED']['OEP_ACCESS']
```
@christian-rli , @jh-RLI - you need to add OEP_ACCESS=OEP (or OEP_DIALECT) in the file config/fred_app.cfg

Bachibouzouk commented on serial_views.py:
```diff
@@ -95,60 +138,80 @@ def ppr_view(request):
+            )
+            # create query
+            if OEP_ACCESS == 'OEP_DIALECT':
+                # TODO find a way to convert the column rea_geom_new to srid 4326
```
@christian-rli - do you know in what srid are the wkb in the table under the column rea_geom_new?

christian-rli replied:
QGIS tells me: srid=3035

Bachibouzouk replied:
Can you find a way to get the transformation between 3035 to 4326 with shapely? *we can't unfortunately make use of ST_Transform of SQLAlchemy as oedialect does not handle it yet :(
